psh/hm/hm.c

```diff
@@ -121,20 +126,149 @@ void psh_hminfo(void)
 }
 
 
+static int psh_hm_cmppid(const void *t1, const void *t2)
+{
+	return (int)((threadinfo_t *)t1)->pid - (int)((threadinfo_t *)t2)->pid;
+}
+
+
+static void psh_hm_freeProcesses(rbtree_t *ptree)
+{
+	rbnode_t *n;
+	proc_t *p;
+
+	for (n = ptree->root; n != NULL; n = ptree->root) {
+		p = lib_treeof(proc_t, node, n);
+		lib_rbRemove(ptree, n);
+		free(p);
+	}
+}
+
+
+static int psh_hm_getProcesses(rbtree_t *ptree)
+{
+	int tcnt, i, j, n = 32;
+	threadinfo_t *info, *rinfo;
+	proc_t *p;
+
+	if ((info = malloc(n * sizeof(threadinfo_t))) == NULL) {
```

Comment:
It would be better to reuse the table if possible and do realloc only if  threadsinfo fail (e.g. pass threadinfo_t **table and it's size (0 for new table) to the function)